spec/system/user_page/user_preferences_security_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+describe "User preferences for Security", type: :system do
+  fab!(:password) { "kungfukenny" }
+  fab!(:email) { "email@user.com" }
+  fab!(:user) { Fabricate(:user, email: email, password: password) }
+  let(:user_preferences_security_page) { PageObjects::Pages::UserPreferencesSecurity.new }
+  let(:user_menu) { PageObjects::Components::UserMenu.new }
+
+  before do
+    user.activate
+    sign_in(user)
+  end
+
+  describe "Security keys" do
+    it "adds a 2F security key and logs in with it" do
+      # simulate browser credential authorization
+      options = ::Selenium::WebDriver::VirtualAuthenticatorOptions.new
+      page.driver.browser.add_virtual_authenticator(options)
+
+      user_preferences_security_page.visit(user)
+      user_preferences_security_page.visit_second_factor(password)
+
+      find(".security-key .new-security-key").click
+      expect(user_preferences_security_page).to have_css("input#security-key-name")
```

Comment:
I'm wondering why we don't use a predicate matcher so that all the logic is encapsulated within the page object which is the purpose.